backend/compact-connect/lambdas/purchases/tests/function/test_handlers/test_credentials.py

```diff
@@ -0,0 +1,125 @@
+        mock_purchase_client_constructor.return_value = mock_purchase_client
+        mock_purchase_client.validate_and_store_credentials.return_value = {
+            'message': 'Successfully verified credentials'
+        }
+
+        return mock_purchase_client
+
+    def _when_purchase_client_raises_exception(self, mock_purchase_client_constructor):
+        mock_purchase_client = MagicMock()
+        mock_purchase_client_constructor.return_value = mock_purchase_client
+        mock_purchase_client.validate_and_store_credentials.side_effect = CCInvalidRequestException(
+            # actual error code and error message from the authorize.net client
+            'Failed to verify credentials. Error code: E00124, Error message: ' 'The provided access token is invalid'
+        )
+
+        return mock_purchase_client
+
+    @patch('handlers.credentials.PurchaseClient')
+    def test_post_payment_processor_credentials_calls_purchase_client_with_expected_parameters(
+        self, mock_purchase_client_constructor
+    ):
+        mock_purchase_client = self._when_purchase_client_successfully_verifies_credentials(
+            mock_purchase_client_constructor
+        )
+        from handlers.credentials import post_payment_processor_credentials
+
+        event = self._when_testing_compact_admin_user()
+
+        resp = post_payment_processor_credentials(event, self.mock_context)
+        self.assertEqual(200, resp['statusCode'])
+
+        purchase_client_call_kwargs = mock_purchase_client.validate_and_store_credentials.call_args.kwargs
+
+        self.assertEqual(TEST_COMPACT, purchase_client_call_kwargs['compact_name'])
+        self.assertEqual(json.loads(event['body']), purchase_client_call_kwargs['credentials'])
+
+    @patch('handlers.credentials.PurchaseClient')
+    def test_post_payment_processor_credentials_returns_exception_message_if_credentials_invalid(
+        self, mock_purchase_client_constructor
+    ):
+        from handlers.credentials import post_payment_processor_credentials
+
+        self._when_purchase_client_raises_exception(mock_purchase_client_constructor)
+
+        event = self._when_testing_compact_admin_user()
+
+        resp = post_payment_processor_credentials(event, self.mock_context)
+        self.assertEqual(400, resp['statusCode'])
+        response_body = json.loads(resp['body'])
+
+        self.assertEqual(
+            {
+                'message': 'Failed to verify credentials. Error code: E00124, Error message: '
+                'The provided access token is invalid'
+            },
+            response_body,
+        )
+
+    @patch('handlers.credentials.PurchaseClient')
+    def test_post_payment_processor_credentials_returns_unauthorized_for_jurisdiction_admins(
+        self, mock_purchase_client_constructor
+    ):
+        from handlers.credentials import post_payment_processor_credentials
+
+        self._when_purchase_client_raises_exception(mock_purchase_client_constructor)
```

Comment:
wouldn't we want this to succeed to test the authorization specifically?

Reply:
Not sure what you mean. Only compact admins should be allowed to call this endpoint, in this case, this is a jurisdiction admin calling the endpoint, not a compact admin, so it should return a 403

Reply:
We have a previous test that verifies that a compact admin is able to successfully call this endpoint and get a 200 response

Reply:
Sorry, I mean to say that `_when_purchase_client_raises_exception` raises an exception for invalid credentials. That is what we intend to test in the prior test, `test_post_payment_processor_credentials_returns_exception_message_if_credentials_invalid`, but that seems incorrect for this test, no?

The authorize.net credentials should be valid, but the post should fail because of the user's permissions.

Reply:
I see, yes I've updated the test to be more clear that the test should be returning 403 since auth failed. I also added a check to make sure the purchase client is never called in this scenario